datenguide_python/query_builder.py

```diff
@@ -351,8 +523,41 @@ def get_fields(self) -> List[str]:
         """
         return self.start_field.get_fields()
 
-    def results(self) -> Optional[ExecutionResults]:
-        return QueryExecutioner().run_query(self)
+    def results(self) -> Union[DataFrame, Optional[ExecutionResults]]:
+        """Runs the query and returns a Pandas DataFrame with the results.
+
+        Raises:
+            ValueError: If the Query did not return any results.
+            E.g. if the Query was ill-formed.
+
+        Returns:
+            Union[DataFrame, Optional[ExecutionResults]] --
+            A DataFrame with the queried data.
+            If the query fails raise ValueError.
+        """
+        result = QueryExecutioner().run_query(self)
+        if result:
+            return QueryOutputTransformer(result.query_results).transform()
```

Comment:
This is definitely how it should end up, but I don't think the query transformer already works like this. Lets keep an eye on this to get a consistent state of the code.